### ISSUE TITLE
Missing include in lv_theme.h

### ISSUE THREAD

**Issue by MiSimon** [lvgl/lvgl#264]
Hi,

in the _lv_theme.h_ file there is no include for _lv_theme_material.h_, this leads to a compiler error if i try to use the material theme.
Adding the include to the other includes at the end of _lv_theme.h_ solves this problem.

Regards,
Michael

**Comment by BesitzeRuf** [comment]
Hi,  lv_theme.h doesn't use anything from lv_theme_material.h. You should analyze more to identify which file requires the lv_theme_material.h...

**Comment by BesitzeRuf** [comment]
Aha, sorry... I double checked the lv_theme.h... and all these includes are at the bottom of the file :D So yes, you're right, it should be included to the lv_theme.h and mby move all #include... to the top?

**Comment by MiSimon** [comment]
The lv_theme_XXX.h files need the definition of lv_theme_t, so the includes need to stay at the bottom of the file.

**Comment by BesitzeRuf** [comment]
There are 2 solutions:

1. Reorganize headers.
2. Use forward declaration:
- a) set some name of struct for lv_theme_t type (for ex. lv_theme_s)
- b) in each lv_theme_XXX.h declare somewhere at the top "typedef struct lv_theme_s lv_theme_t; "


<details> 
  <summary>lv_theme.h</summary>

> 
> /**
>  *@file lv_themes.h
>  *
>  */
> 
> #ifndef LV_THEMES_H
> #define LV_THEMES_H
> 
> #ifdef __cplusplus
> extern "C" {
> #endif
> 
> /*********************
>  *    INCLUDES
>  *********************/
> #include "../../lv_conf.h"
> #include "../lv_core/lv_style.h"
> 
> **#include "lv_theme_templ.h"**
> **#include "lv_theme_default.h"**
> **#include "lv_theme_alien.h"**
> **#include "lv_theme_night.h"**
> **#include "lv_theme_zen.h"**
> **#include "lv_theme_mono.h"**
> **#include "lv_theme_material.h"**
> 
> /*********************
>  *    DEFINES
>  *********************/
> 
> /**********************
>  *    TYPEDEFS
>  **********************/
> 
> typedef struct lv_theme_s{
>     lv_style_t *bg;
>     lv_style_t *panel;
> 
> #if USE_LV_CONT != 0
>     lv_style_t *cont;
> #endif
> #if USE_LV_BTN != 0
>     struct {
>         lv_style_t *rel;
>         lv_style_t *pr;
>         lv_style_t *tgl_rel;
>         lv_style_t *tgl_pr;
>         lv_style_t *ina;
>     }btn;
> #endif
> 
> #if USE_LV_LABEL != 0
>     struct {
>         lv_style_t *prim;
>         lv_style_t *sec;
>         lv_style_t *hint;
>     }label;
> #endif
> 
> #if USE_LV_IMG != 0
>     struct {
>         lv_style_t *light;
>         lv_style_t *dark;
>     }img;
> #endif
> 
> #if USE_LV_LINE != 0
>     struct {
>         lv_style_t *decor;
>     }line;
> #endif
> 
> #if USE_LV_LED != 0
>     lv_style_t *led;
> #endif
> 
> #if USE_LV_BAR != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *indic;
>     }bar;
> #endif
> 
> #if USE_LV_SLIDER != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *indic;
>         lv_style_t *knob;
>     }slider;
> #endif
> 
> #if USE_LV_LMETER != 0
>     lv_style_t *lmeter;
> #endif
> 
> #if USE_LV_GAUGE != 0
>     lv_style_t *gauge;
> #endif
> 
> #if USE_LV_SW != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *indic;
>         lv_style_t *knob_off;
>         lv_style_t *knob_on;
>     }sw;
> #endif
> 
> #if USE_LV_CHART != 0
>     lv_style_t *chart;
> #endif
> 
> #if USE_LV_CB != 0
>     struct {
>         lv_style_t *bg;
>         struct {
>             lv_style_t *rel;
>             lv_style_t *pr;
>             lv_style_t *tgl_rel;
>             lv_style_t *tgl_pr;
>             lv_style_t *ina;
>         }box;
>     }cb;
> #endif
> 
> #if USE_LV_BTNM != 0
>     struct {
>         lv_style_t *bg;
>         struct {
>             lv_style_t *rel;
>             lv_style_t *pr;
>             lv_style_t *tgl_rel;
>             lv_style_t *tgl_pr;
>             lv_style_t *ina;
>         }btn;
>     }btnm;
> #endif
> 
> #if USE_LV_KB != 0
>     struct {
>         lv_style_t *bg;
>         struct {
>             lv_style_t *rel;
>             lv_style_t *pr;
>             lv_style_t *tgl_rel;
>             lv_style_t *tgl_pr;
>             lv_style_t *ina;
>         } btn;
>     }kb;
> #endif
> 
> #if USE_LV_MBOX != 0
>     struct {
>         lv_style_t *bg;
>         struct {
>             lv_style_t *bg;
>             lv_style_t *rel;
>             lv_style_t *pr;
>         }btn;
>     }mbox;
> #endif
> 
> #if USE_LV_PAGE != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *scrl;
>         lv_style_t *sb;
>     }page;
> #endif
> 
> #if USE_LV_TA != 0
>     struct {
>         lv_style_t *area;
>         lv_style_t *oneline;
>         lv_style_t *cursor;
>         lv_style_t *sb;
>     }ta;
> #endif
> 
> #if USE_LV_LIST
>     struct {
>         lv_style_t *bg;
>         lv_style_t *scrl;
>         lv_style_t *sb;
>         struct {
>             lv_style_t *rel;
>             lv_style_t *pr;
>             lv_style_t *tgl_rel;
>             lv_style_t *tgl_pr;
>             lv_style_t *ina;
>         }btn;
>     }list;
> #endif
> 
> #if USE_LV_DDLIST != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *sel;
>         lv_style_t *sb;
>     }ddlist;
> #endif
> 
> #if USE_LV_ROLLER != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *sel;
>     }roller;
> #endif
> 
> #if USE_LV_TABVIEW != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *indic;
>         struct {
>             lv_style_t *bg;
>             lv_style_t *rel;
>             lv_style_t *pr;
>             lv_style_t *tgl_rel;
>             lv_style_t *tgl_pr;
>         }btn;
>     }tabview;
> #endif
> 
> #if USE_LV_WIN != 0
>     struct {
>         lv_style_t *bg;
>         lv_style_t *sb;
>         lv_style_t *header;
>         struct {
>             lv_style_t *bg;
>             lv_style_t *scrl;
>         }content;
>         struct {
>             lv_style_t *rel;
>             lv_style_t *pr;
>         }btn;
>     }win;
> #endif
> }lv_theme_t;
> 
> /**********************
>  *  GLOBAL PROTOTYPES
>  **********************/
> 
> /**
>  * Set a theme for the system.
>  * From now, all the created objects will use styles from this theme by default
>  * @param th pointer to theme (return value of: 'lv_theme_init_xxx()')
>  */
> void lv_theme_set_current(lv_theme_t *th);
> 
> /**
>  * Get the current system theme.
>  * @return pointer to the current system theme. NULL if not set.
>  */
> lv_theme_t * lv_theme_get_current(void);
> 
> /**********************
>  *    MACROS
>  **********************/
> 
> /**********************
>  *     POST INCLUDE
>  *********************/
> 
> 
> #ifdef __cplusplus
> } /* extern "C" */
> #endif
> 
> #endif /*LV_THEMES_H*/
> 


</details>

<details> 
  <summary>lv_theme_material.h</summary>
 
>   /**
>  * @file lv_theme_material.h
>  *
>  */
> 
> #ifndef LV_THEME_MATERIAL_H
> #define LV_THEME_MATERIAL_H
> 
> #ifdef __cplusplus
> extern "C" {
> #endif
> 
> /*********************
>  *      INCLUDES
>  *********************/
> #include "../../lv_conf.h"
> 
> #if USE_LV_THEME_MATERIAL
> 
> /*********************
>  *      DEFINES
>  *********************/
> 
> /**********************
>  *      TYPEDEFS
>  **********************/
> typedef struct lv_theme_s lv_theme_t; // forward declaration
> /**********************
>  * GLOBAL PROTOTYPES
>  **********************/
> 
> /**
>  * Initialize the material theme
>  * @param hue [0..360] hue value from HSV color space to define the theme's base color
>  * @param font pointer to a font (NULL to use the default)
>  * @return pointer to the initialized theme
>  */
> lv_theme_t * lv_theme_material_init(uint16_t hue, lv_font_t *font);
> 
> /**
>  * Get a pointer to the theme
>  * @return pointer to the theme
>  */
> lv_theme_t * lv_theme_get_material(void);
> 
> /**********************
>  *      MACROS
>  **********************/
> 
> #endif
> 
> #ifdef __cplusplus
> } /* extern "C" */
> #endif
> 
> #endif /*LV_THEME_MATERIAL_H*/
> 

</details>

**Comment by MiSimon** [comment]
I created a pull request for the missing include.
The reorganization of the header files is not directly related to this issue, so i did not include it in the pr.

**Comment by kisvegabor** [comment]
Hi,

I added includes for all themes in `lv_theme.h` to enable the inclusion of all themes at once.

@MiSimon Thank you for highlighting the missing include. I merged your Pull Request.

@BesitzeRuf Thank your for sharing your thoughts! I considered to use of forward declaration. It really would be a more elegant and professional solution. However, as the current organization is quite simple and not requires any additional declaration in the lv_theme_XXX files I would stay with that.
